### PR TITLE
2 local aws filesync shell mod

### LINKED ISSUE #10: Dashboard: EN Search Query analytics - most used queries or search terms
### Checked for duplicates

No - I haven't checked

### 🧑‍🔬 User Persona(s)

_No response_

### 💪 Motivation

_No response_

### 📖 Additional Details

Dashboard showing most commonly used search terms.

### Acceptance Criteria

_No response_

### ⚙️ Engineering Details

_No response_

### LINKED ISSUE #15: Create sync script for syncing logs from logging server to AWS and load into Athena

### PR DESCRIPTION
**Pull request by kaipak**
Complete dev log wildcard list.

resolves #10
resolves #15

### CODE REVIEW

**Comment by jordanpadams** [comment]
@kaipak can you take a look at the things caught by sonatype above?

File: -arlt [hunk 1]
```diff
@@ -0,0 +1,76 @@
+diff --git a/config/def.yaml b/config/def.yaml
```
Comment (nutjob4life) [inline -arlt:1]: What is this file? 

It looks like someone ran the output of `git diff` into a file named `-arlt` and included it as part of the commit.

Is this something required by AWS? I'm not sure about this. If this was checked-in by mistake, let's delete `-arlt` from the branch.

File: config/config_dev.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+log_directory: /report_service/logs/final
+#log_directory: /home/kaipak/datasets/report_service/logs/final
```
Comment (nutjob4life) [inline config/config_dev.yaml:2]: I recommend removing commented-out lines; they usually lead to suspicion or confusion ("Uh oh was this really needed?", that sort of thing 😁)

If they should stay, they include an _additional_ comment explaining why ("commenting this out but it is useful to put back in during development", that sort of thing 😅)

File: config/config_dev.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+profile_name: saml-pub
+s3_bucket: pds-web-analytics-cloud
+s3_logdir: logs_dev
+#s3_logdir: logs_dev
```
Comment (nutjob4life) [inline config/config_dev.yaml:6]: In this case, the commented-out line is identical to the line above it, so definitely delete this line

File: config/config_prod.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+log_directory: /report_service/logs/final
+#log_directory: /home/kaipak/datasets/report_service/logs/final
```
Comment (nutjob4life) [inline config/config_prod.yaml:2]: Ditto

File: config/config_prod.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+profile_name: saml-pub
+s3_bucket: pds-web-analytics-cloud
+s3_logdir: logs
+#s3_logdir: logs_dev
```
Comment (nutjob4life) [inline config/config_prod.yaml:6]: Ah this time the `s3_logdir` is `logs`, not `logs_dev` as in the above file. Let's double-check this.

File: scripts/s3_log_sync.py [hunk 6]
```diff
@@ -3,11 +3,113 @@
...
+        self.s3_sync_cmd += ["--exclude", "*"]
 
+    def run(self):
+        for src_path in self.src_paths.items():
+            self.sync_directory(src_path)
 
-s3 = boto3.resource('s3')
+    def sync_directory(self, path_tuple):
+        # Use subprocess to call the AWS CLI command and display progress
+        # HACK
```
Comment (nutjob4life) [inline scripts/s3_log_sync.py:38]: I like the honesty here 😅

File: scripts/s3_log_sync.py [hunk 7]
```diff
@@ -3,11 +3,113 @@
...
+    @staticmethod
+    def convert_size(size):
+        # Convert bytes to a human-readable format
+        if size == 0:
+            return "0B"
+        size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+        i = int(math.floor(math.log(size, 1024)))
+        p = math.pow(1024, i)
+        s = round(size / p, 2)
+        return f"{s}{size_name[i]}"
```
Comment (nutjob4life) [inline scripts/s3_log_sync.py:80]: The Python [humanize](https://pypi.org/project/humanize/) API provides convenient functions to do this for you; it even goes up to YB

File: scripts/s3_log_sync.py [hunk 8]
```diff
@@ -3,11 +3,113 @@
...
+    def get_throughput(sent, start_time):
+        # Calculate and return the current throughput in MB/s
+        elapsed_time = time.monotonic() - start_time
+        mb_sent = sent / (1024 * 1024)
+        return f"{mb_sent / elapsed_time:.2f} MB/s"
+
+
+if __name__ == '__main__':
+    # Parse basic config file and set up AWS Session
+    with open('../config/config_dev.yaml', 'r') as file:
```
Comment (nutjob4life) [inline scripts/s3_log_sync.py:104]: Maybe a command-line option to provide different `.yaml` files as input?

**Review by nutjob4life — CHANGES_REQUESTED** [review]

**Comment by kaipak** [comment]
> @kaipak can you take a look at the things caught by sonatype above?

Yep, will do. Have been away on travel and conferences the last couple of weeks.

File: -arlt (hunk 1 above)
Comment (kaipak) [reply -arlt]: Removed.

File: scripts/s3_log_sync.py (hunk 7 above)
Comment (kaipak) [reply scripts/s3_log_sync.py:82]: Will include in next PR.

File: scripts/s3_log_sync.py (hunk 8 above)
Comment (kaipak) [reply scripts/s3_log_sync.py]: Will include in next PR.

File: src/clf_data_augmenter.py [hunk 9]
```diff
@@ -6,30 +6,35 @@
...
 
         # Assume if I can't categorize the tool, it's just basic content.
-        self.df['tool'] = "content"
-        self.df.loc[(~self.df['log_name'].astype('str').str.contains("tool", na=False)) &
-                   (self.df['request'].astype('str').str.contains('viewer|tracker|ephem')), 'tool'] = (
-            self.df['request'].str.split(r'/').str.get(2).str.split('_').str.get(0)
+        self.df["tool"] = "content"
+        self.df.loc[
+            (~self.df["log_name"].astype("str").str.contains("tool", na=False))
+            & (self.df["request"].astype("str").str.contains("viewer|tracker|ephem")),
```
[the remark this answers is not in the document]
Comment (kaipak) [reply src/clf_data_augmenter.py]: file removed.

File: scripts/s3_log_sync.py [hunk 10]
```diff
@@ -3,11 +3,118 @@
...
+        # HACK
+        src_path = path_tuple[0]
+        path_include = path_tuple[1]["include"]
+        s3_path = os.path.join(self.s3_subdir, os.path.relpath(src_path, self.src_logdir))
+        cmd = self.s3_sync_cmd + [src_path, f"s3://{self.bucket_name}/{s3_path}"]
+        for include in path_include:
+            cmd += ["--include", include]
+        start_time = time.monotonic()
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                              bufsize=1, universal_newlines=True, shell=False) as proc:
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply scripts/s3_log_sync.py:45]: @kaipak ☝️

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**